File: web-socket-sse/be/src/main/java/org/rookedsysc/be/model/GameEvent.java
package org.rookedsysc.be.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameEvent {

    public enum EventType {
        SHOT, FOUL, EJECTION
    }

    private EventType type;
    private String playerId;
    private String playerName;
    private String teamId;
    private String teamName;
    private LocalDateTime timestamp;
    private String description;
    private Boolean successful; // For shots - whether it went in
    private Integer points; // For shots - number of points
} 

File: web-socket-sse/fe/src/types/GameEvent.ts
export enum EventType {
  SHOT = "SHOT",
  FOUL = "FOUL",
  EJECTION = "EJECTION",
}

export interface GameEvent {
  type: EventType;
  playerId: string;
  playerName: string;
  teamId: string;
  teamName: string;
  timestamp?: string; // using string for easier form handling, will convert to LocalDateTime
  description: string;
  successful?: boolean; // For shots - whether it went in
  points?: number; // For shots - number of points
}

export const createEmptyGameEvent = (): GameEvent => ({
  type: EventType.SHOT,
  playerId: "",
  playerName: "",
  teamId: "",
  teamName: "",
  description: "",
  successful: false,
  points: 0,
});
